pkg/services/object/put/distributed.go

```diff
@@ -224,6 +203,36 @@ func (t *distributedTarget) Close() (oid.ID, error) {
 	return id, nil
 }
 
+func (t *distributedTarget) getShardObjectMetadata() []byte {
+	if t.objSharedMeta != nil {
```

Comment:
when this can happen?

Reply:
it was an attempt to do encoding only when necessary. Separate topic, besides, it requires reset and synchronization, which I forgot about, so reverted for now